tests/test_core.py

```diff
@@ -2419,6 +2419,16 @@ def test_pthread_specific(self):
   def test_pthread_equal(self):
     self.do_run_in_out_file_test('pthread/test_pthread_equal.cpp')
 
+  @node_pthreads
+  def test_pthread_proxying(self):
+    self.set_setting('EXIT_RUNTIME')
+    self.set_setting('PROXY_TO_PTHREAD')
+    self.set_setting('PTHREAD_POOL_SIZE=3')
+    self.set_setting('INITIAL_MEMORY=32mb')
+    self.set_setting('EXPORTED_FUNCTIONS=_emscripten_proxy_execute_queue,_main')
```

Comment:
If you mark `_emscripten_proxy_execute_queue` as `EMSCRIPTEN_KEEPALIVE` I think it will be exported if and only if proxying.o is included, which seems like what you want (then you can delete this line I think).

Perhaps add a comment saying `Exporting so it can be called by worker.js`.

Reply:
I will make that minimal change, but I wonder whether the other functions should be similarly exported since they're designed to be called on arbitrary threads as well.

Reply:
The "exporting" we are talking about is only needed for the case when external JS code (e.g. user code, or worker.js) needs to call a function directly.

Reply:
Oh I see. Should we add it to the list here then? https://github.com/emscripten-core/emscripten/blob/main/emcc.py#L1971-L1982

Reply:
We could.. but adding `EMSCRIPTEN_KEEPALIVE` the the source code would allow it to be only exported if the object file is used.. which would avoid paying the cost in all pthread-based programs.

Even better you can do `if hasExportedFunction('emscripten_proxy_execute_queue')` in the library code (and the worker.js code) to avoid included the message handlers unless they are needed.